src/data/mystery-encounter-option.ts

```diff
@@ -20,15 +26,57 @@ export default class MysteryEncounterOption implements MysteryEncounterOption {
   meetsRequirements?(scene: BattleScene) {
     return !this.requirements.some(requirement => !requirement.meetsRequirement(scene));
   }
+
+  meetsProtagonistRequirementAndProtagonistPokemonSelected?(scene: BattleScene) {
```

Comment:
note to self, this probably needs to be matched with mystery-encounter.ts line 150. could refactor. 
Need to reconsider the cases in which you'd query for a new protagonist for an option midway through the event...

Reply:
i just copied and pasted this to match. we should probably refactor it somehow but... i'm lazy